BlockchainSdk/BlockchainService/TON/TONProviderProvider/TONProvider.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  TONProvider.swift
+//  BlockchainSdk
+//
+//  Created by skibinalexander on 26.01.2023.
+//  Copyright © 2023 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+struct TONProvider: HostProvider {
+    
+    // MARK: - HostProvider
+    
+    /// Blockchain API host
+    var host: String {
+        node.endpoint.url.path
+    }
+    
+    /// Configuration connection node for provider
+    private(set) var node: TONNetworkNode
+    
+    // MARK: - Properties
+    
+    /// Network provider of blockchain
+    private(set) var network: NetworkProvider<TONProviderTarget>
+    
+    // MARK: - Init
+    
+    init?(
+        nodeName: TONNodeName,
+        config: BlockchainSdkConfig,
+        network: NetworkProvider<TONProviderTarget>,
+        isTestnet: Bool
+    ) {
+        var apiKeyValue: String?
```

Comment:
```suggestion
        let apiKeyValue: String
```
можно так? и можно будет проверки убрать